File: sdk/src/main/java/org/maibot/sdk/model/APIResponse.java
package org.maibot.sdk.model;

import com.openai.models.responses.Response;

public record APIResponse(
  Response response,
  TokenStatistics tokenStatistics
) {
    public record TokenStatistics(
      long inputTokens,
      long outputTokens,
      long totalTokens
    ) {
    }
}

File: sdk/src/main/java/org/maibot/sdk/model/APIResponse.kt
package org.maibot.sdk.model

import org.maibot.sdk.model.payload.MessageContextItem
import tools.jackson.databind.JsonNode

@JvmRecord
data class APIResponse(
    val oriResp: Any,
    val reasoning: String?,
    val response: MessageContextItem?,
    val toolCalls: List<ToolCallInfo>?,
    val tokenStatistics: TokenStatistics?
) {
    data class ToolCallInfo(
        val toolName: String,
        val toolArgs: JsonNode?,
        val id: String?
    )

    @JvmRecord
    data class TokenStatistics(
        val inputTokens: Long,
        val outputTokens: Long,
        val totalTokens: Long
    )
}
